src/SprykerEco/Zed/Econda/Business/EcondaBusinessFactory.php

```diff
@@ -0,0 +1,285 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\Econda\Business;
+
+use Exception;
+use Generated\Shared\Transfer\BatchResultTransfer;
+use Generated\Shared\Transfer\FailedResultTransfer;
+use Spryker\Shared\SqlCriteriaBuilder\CriteriaBuilder\CriteriaBuilderDependencyContainer;
+use Spryker\Shared\SqlCriteriaBuilder\CriteriaBuilder\CriteriaBuilderFactory;
+use Spryker\Shared\SqlCriteriaBuilder\CriteriaBuilder\CriteriaBuilderFactoryWorker;
+use Spryker\Zed\Kernel\Business\AbstractBusinessFactory;
+use SprykerEco\Zed\Econda\Business\Collector\File\EcondaCategoryCollector;
+use SprykerEco\Zed\Econda\Business\Collector\File\EcondaProductCollector;
+use SprykerEco\Zed\Econda\Business\Exporter\FileExporter;
+use SprykerEco\Zed\Econda\Business\Exporter\Runner;
+use SprykerEco\Zed\Econda\Business\Exporter\Writer\File\Adapter\CsvAdapter;
+use SprykerEco\Zed\Econda\Business\Exporter\Writer\File\FileWriter;
+use SprykerEco\Zed\Econda\Business\Exporter\Writer\File\NameGenerator\CsvNameGenerator;
+use SprykerEco\Zed\Econda\Business\Helper\ProgressBarHelper;
+use SprykerEco\Zed\Econda\Business\Manager\CollectorManager;
+use SprykerEco\Zed\Econda\Business\Reader\EcondaCsvFileReader;
+use SprykerEco\Zed\Econda\EcondaDependencyProvider;
+use SprykerEco\Zed\Econda\Persistence\EcondaQueryContainer;
+
+/**
+ * @method \SprykerEco\Zed\Econda\EcondaConfig getConfig()
+ * @method \SprykerEco\Zed\Econda\Persistence\EcondaQueryContainer getQueryContainer()
+ */
+class EcondaBusinessFactory extends AbstractBusinessFactory
+{
+
+    /**
+     * @param string $type
+     * @param string $locale
+     *
+     * @return string
+     */
+    public function getEcondaCsvFileContent($type, $locale)
+    {
+        $nameGenerator = $this->createCsvNameGenerator();
+        $reader = new EcondaCsvFileReader($this->getConfig(), $nameGenerator);
+        return $reader->readFile($type, $locale);
```

Comment:
This looks like the factory is abused for model logic, the factory should only instance classes however.
So you return the file reader class object and then run readFile() on it from the invoking code.